service/src/main/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStep.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.resource.controlled.ControlledResourceService;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * A step to delete all controlled Azure resources resources in a workspace. This reads the list of
+ * controlled Azure resources in a workspace from the WSM database.
+ */
+public class DeleteControlledAzureResourcesStep implements Step {
+
+  private final Logger logger = LoggerFactory.getLogger(DeleteControlledAzureResourcesStep.class);
+  private final ResourceDao resourceDao;
+  private final ControlledResourceService controlledResourceService;
+  private final UUID workspaceUuid;
+  private final AuthenticatedUserRequest userRequest;
+
+  public DeleteControlledAzureResourcesStep(
+      ResourceDao resourceDao,
+      ControlledResourceService controlledResourceService,
+      UUID workspaceUuid,
+      AuthenticatedUserRequest userRequest) {
+    this.resourceDao = resourceDao;
+    this.controlledResourceService = controlledResourceService;
+    this.workspaceUuid = workspaceUuid;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+
+    List<ControlledResource> controlledResourceList =
+        resourceDao.listControlledResources(workspaceUuid, CloudPlatform.AZURE);
```

Comment:
Are there any dependencies between resources that need to be considered?

Reply:
The only dependency I'm aware of is a Storage container on a Storage account, but I'll look more closely at the existing Azure resources to be more certain

Reply:
Hmm... I thought the disk delete would fail if it were mounted on a running VM, but you would know better than me.

Reply:
Yeah that's a great point. Also not sure what happens if we delete a VM out from under an Azure IP for example. I misspoke on my first comment -- the only dependency I was aware of \*off the top of my head\* was the container -> account dependency. There are other dependencies that will need to be considered